src/backend/joanie/core/serializers/client.py

```diff
@@ -1233,6 +1241,52 @@ def create(self, validated_data):
             organization = get_object_or_404(models.Organization, id=organization_id)
             validated_data["organization"] = organization
 
+        try:
+            course_id = validated_data["course"].id
+        except KeyError:
+            course_id = validated_data["enrollment"].course_run.course_id
+
+        try:
+            product_id = validated_data["product"].id
+        except KeyError:
+            product_id = validated_data["enrollment"].course_run.product_id
+
+        filters = {"product": product_id, "course": course_id}
+        if organization_id:
+            filters.update({"organizations": organization_id})
+        course_product_relation = models.CourseProductRelation.objects.get(**filters)
+
+        order_groups = models.OrderGroup.objects.find_actives(
+            course_product_relation_id=course_product_relation.id
+        )
+        seats_limitation = None
+        for order_group in order_groups:
+            if order_group.nb_seats is not None:
+                if order_group.available_seats == 0:
+                    seats_limitation = order_group
+                    continue
+
+                seats_limitation = None
+
+            if order_group.is_enabled:
+                if "order_groups" not in validated_data:
+                    validated_data["order_groups"] = []
+                validated_data["order_groups"].append(order_group)
+
+        if seats_limitation:
+            raise serializers.ValidationError(
+                {
+                    "order_group": [
+                        f"Maximum number of orders reached for "
+                        f"product {validated_data['product'].title:s}"
+                    ]
+                }
+            )
+
+        if voucher_code := self.initial_data.get("voucher_code"):
+            voucher = models.Voucher.objects.get(code=voucher_code)
+            validated_data["voucher_id"] = voucher.id
```

Comment:
We should decorate the serializer to make drf spectacular aware of this property.